client/pages/Event/EventDetailsPage.jsx

```diff
@@ -186,46 +206,66 @@ class EventDetailsPage extends React.Component {
   };
 
   handleBack() {
-    const { history: { push } } = this.props;
+    const {
+      history: { push }
+    } = this.props;
     push('/dashboard');
   }
 
   loadEvent() {
-    const { match: { params: { eventId } } } = this.props;
+    const {
+      match: {
+        params: { eventId }
+      }
+    } = this.props;
     const { getEventAction } = this.props;
     getEventAction(eventId);
   }
 
+  showSlackChannels = evt => {
+    evt.preventDefault();
+    this.setState(
+      {
+        showSlackChannels: true
+      },
+      () => {
+        document.addEventListener('click', this.closeChannelsMenu);
```

Comment:
Why are we manually adding event listeners to the DOM here?

Reply:
I did that to handle the interaction of clicking outside the menu to close the menu options

Reply:
Ok cool. We have a package installed which may be able to handle that for you. See `react-onclickoutside`

Reply:
wow... okay I'll check it out

Reply:
thanks